kivy/lib/osc/oscAPI.py

```diff
@@ -80,29 +81,42 @@ def _queue_message(self, message):
 
         def _get_isRunning(self):
             return self._isRunning.value
+
         def _set_isRunning(self, value):
             self._isRunning.value = value
         isRunning = property(_get_isRunning, _set_isRunning)
 
         def _get_haveSocket(self):
             return self._haveSocket.value
+
         def _set_haveSocket(self, value):
             self._haveSocket.value = value
         haveSocket = property(_get_haveSocket, _set_haveSocket)
 else:
     def _readQueue(thread_id=None):
-        pass
+        for id in oscThreads:
+                if thread_id is not None:
+                    if id != thread_id:
+                        continue
+                thread = oscThreads[id]
+
+        q = thread.queue
+        h = thread.addressManager.handle
+
+        while q:
+            h(q.popleft())
```

Comment:
i guess one lookup could be saved by doing p = thread.queue.popleft and using h(p()), since it could be a tightloop, maybe that's better?